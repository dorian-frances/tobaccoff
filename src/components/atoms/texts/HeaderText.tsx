import { Text, StyleSheet } from 'react-native';
import React from 'react';

type TextHeaderProps = {
  text: string;
};

const HeaderText = ({ text = 'Lorem Ipsum' }: TextHeaderProps) => {
  return <Text style={styles.text}>{text}</Text>;
};

const styles = StyleSheet.create({
  text: {
    fontFamily: 'Montserrat-Bold',
    fontSize: 40,
  },
});

export default HeaderText;
