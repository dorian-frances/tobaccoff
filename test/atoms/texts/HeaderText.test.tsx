import React from 'react';
import HeaderText from '../../../src/components/atoms/texts/HeaderText';
import { render } from '@testing-library/react-native';

describe('<HeaderText/>', () => {
  const fakerText = 'fakeText';
  it('should render correctly', () => {
    const headerText = render(<HeaderText text={fakerText} />);
    expect(headerText.toJSON()).toMatchSnapshot();
  });

  it('should access text', () => {
    const headerText = render(<HeaderText text={fakerText} />);
    expect(headerText.getByText(fakerText)).toBeTruthy();
  });
});
