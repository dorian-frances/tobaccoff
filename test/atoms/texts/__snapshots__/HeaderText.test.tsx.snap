// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderText/> should render correctly 1`] = `
<Text
  style={
    {
      "fontFamily": "Montserrat-Bold",
      "fontSize": 40,
    }
  }
>
  fakeText
</Text>
`;
